src/Diagnostics/Debug/StateCheck.jl

```diff
@@ -0,0 +1,556 @@
+"""
+ StateCheck :: Module with a minimal set of functions for gettings statistics 
+               and basic I/O from ClimateMachine DG state arrays (MPIStateArray type). 
+               Created for regression testing and code change tracking and debugging.
+               StateCheck functions iterate over named variables in an MPIStateArray,
+               calculate and report their statistics and/or write values for all or
+               some subset of points at a fixed frequency.
+
+               Functions:
+               StateCheck.sccreate()   - Create a StateCheck call back variable.
+
+               StateCheck.scdocheck()  - Check Statecheck variable values against reference values.
+
+               StateCheck.scprintref() - Print Statecheck variable in format for creating 
+                                         reference values.
+"""
+module StateCheck
+
+# Imports from standard Julia packages
+using Formatting
+using MPI
+using Printf
+using StaticArrays
+using Statistics
+
+# Imports from ClimateMachine core
+import ClimateMachine.GenericCallbacks: EveryXSimulationSteps
+import ClimateMachine.MPIStateArrays: MPIStateArray
+import ClimateMachine.VariableTemplates: flattenednames
+
+####
+# For testing put a new function signature here!
+# Needs to go in src/Utilities/VariableTemplates/var_names.jl
+# This handles SMatrix case
+flattenednames(::Type{T}; prefix = "") where {T <: SArray} =
```

Comment:
Can we add proper documentation for what this is doing? What's written will not show up in doc strings.